AutoCollection/Statsbeat.ts

```diff
@@ -0,0 +1,263 @@
+    }
+
+    public countRetry() {
+        if (!this.isEnabled()) {
+            return;
+        }
+        this._retryCount++;
+    }
+
+    public async trackStatsbeatMetrics() {
+        this._getResourceProvider(async () => {
+            this._trackRequestDuration();
+            this._trackRequestsCount();
+            await this._sendStatsbeats();
+        });
+    }
+
+    private _trackRequestDuration() {
+        var lastRequests = this._lastRequests;
+        var requests = {
+            totalRequestCount: this._totalRequestCount,
+            time: +new Date
+        };
+        var intervalRequests = (requests.totalRequestCount - lastRequests.totalRequestCount) || 0;
+        var elapsedMs = requests.time - lastRequests.time;
+        var averageRequestExecutionTime = ((this._intervalRequestExecutionTime - this._lastIntervalRequestExecutionTime) / intervalRequests) || 0; // default to 0 in case no requests in this interval
+        this._lastIntervalRequestExecutionTime = this._intervalRequestExecutionTime; // reset
+        if (elapsedMs > 0 && intervalRequests > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_DURATION, value: averageRequestExecutionTime });
+        }
+        this._lastRequests = requests;
+    }
+
+    private _trackRequestsCount() {
+        if (this._totalSuccesfulRequestCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_SUCCESS, value: this._totalSuccesfulRequestCount });
+            this._totalSuccesfulRequestCount = 0; //Reset
+        }
+        if (this._totalFailedRequestCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_FAILURE, value: this._totalFailedRequestCount });
+            this._totalFailedRequestCount = 0; //Reset
+        }
+        if (this._retryCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.RETRY_COUNT, value: this._retryCount });
+            this._retryCount = 0; //Reset
+        }
+        if (this._throttleCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.THROTTLE_COUNT, value: this._throttleCount });
+            this._throttleCount = 0; //Reset
+        }
+        if (this._exceptionCount > 0) {
+            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.EXCEPTION_COUNT, value: this._exceptionCount });
+            this._exceptionCount = 0; //Reset
+        }
+    }
+
+    private async _sendStatsbeats() {
+        let envelopes: Array<Contracts.Envelope> = [];
+        let properties = {
+            "os": this._os,
+            "rp": this._resourceProvider,
+            "cikey": this._cikey,
+            "runtimeVersion": this._runtimeVersion,
+            "language": this._language,
+            "version": this._sdkVersion,
+            "attach": this._attach,
+            "instrumentation": this._instrumentations,
+            "feature": this._features,
```

Comment:
feature is sent every 24 hours.  it's not part of the network statsbeat. please check the design doc: https://bit.ly/3zEf3Vx

Reply:
Feature statbeat is not part of this PR I'm planning to include in other PR adding attach one as well. can you confirm if attach is correct here in network stats?

Reply:
then please don't add feature in the network statsbeat payload. feature is on its own.

Reply:
Must be removed now